### ISSUE TITLE
Should we report old versions of Javascript libraries? How frequently are they exploitable?

### ISSUE THREAD

**Issue by kazet** [CERT-Polska/Artemis#800]

**Comment by RasenRhino** [comment]
ideally we should. we do so . there are a lot of malicious packages out there. you can have a look at [this ](https://security.snyk.io/vuln/npm) . There are a few utilities out there that can be integrated in Artemis

**Comment by christinathucanh** [comment]
Yes, it's crucial to prioritize security patches and stay vigilant against emerging threats in the JavaScript ecosystem. I think while reporting old versions of JavaScript libraries is important for maintaining security and performance, the exploitability of these outdated versions can vary based on factors such as library popularity, severity of vulnerabilities, and timely application of updates by developers.

**Comment by christinathucanh** [comment]
You can refer to https://nvd.nist.gov/, where we can access detailed vulnerability descriptions, affected versions, and potential impact assessments for JavaScript libraries. This data source can help prioritize updates, assess risk levels, and stay informed about emerging threats in the JavaScript ecosystem.

**Comment by kazet** [comment]
I think a good implementation of this task should contain some kind of risk assessment, not just "your jquery version is obsolete" so that we can decrease spam

**Comment by kazet** [comment]
CC: @ap74062 as you're implementing a module that could detect that

**Comment by kazet** [comment]
e.g. CVE-2020-11023

**Comment by HarshitPal25** [comment]
hey hi @kazet umm its a nice idea to inlcude all versions of javascript libraries as i have also listed it in my gsoc proposal to include all versions of JS libraries and i think i can work on that if you like you can include it in the ARTEMIS project.
